### PR TITLE
Jpk/logging tec

### PR DESCRIPTION
**Pull request by jpknoll**

### CODE REVIEW

File: Keas.Core/Services/IEmailService.cs [hunk 1]
```diff
@@ -70,200 +51,145 @@ public async Task SendExpiringMessage(int personId, ExpiringItemsEmailModel mode
...
+                .Select(u => new Recipient() {Address = new Address(u.Email, u.Name, "cc")})
+                .ToList();
+
+#if !DEBUG
+            // add emails to
+            ccEmails.ForEach(transmission.Recipients.Add);
+#endif
+
+            // build view
+            var engine = GetRazorEngine();
```
Comment (jSylvestre) [inline Keas.Core/Services/IEmailService.cs]: @jpknoll Did you test this? We had to jump through some hoops using RazorLightEngineBuilder.

File: Keas.Core/Services/IEmailService.cs [hunk 2]
```diff
@@ -70,200 +51,145 @@ public async Task SendExpiringMessage(int personId, ExpiringItemsEmailModel mode
...
-                }
-                else if (access.ExpiresAt > DateTime.UtcNow.AddDays(1))
-                {
-                    access.NextNotificationDate = access.ExpiresAt.AddDays(-1);
-                }
-                else
-                {
-                    access.NextNotificationDate = DateTime.UtcNow.AddDays(1);
-                }
-                _dbContext.AccessAssignments.Update(access);
```
Comment (jSylvestre) [inline Keas.Core/Services/IEmailService.cs]: @jpknoll Was the update statement not needed?

File: Keas.Core/Services/IEmailService.cs [hunk 3]
```diff
@@ -272,20 +198,35 @@ public async Task SendNotificationMessage(User user)
                 _dbContext.Notifications.Update(notification);                
             }
             await _dbContext.SaveChangesAsync();
+        }
 
-            var mimeType = new System.Net.Mime.ContentType("text/html");
-
-            var alternate = AlternateView.CreateAlternateViewFromString(message.Body, mimeType);
-            message.AlternateViews.Add(alternate);
```
Comment (jSylvestre) [inline Keas.Core/Services/IEmailService.cs]: @jpknoll Does Sparkpost have AlternateViews?

File: Keas.Core/Services/IEmailService.cs (hunk 2 above)
Comment (jpknoll) [reply Keas.Core/Services/IEmailService.cs]: No, the objects are already in tracking. Just need the save at the end.

File: Keas.Core/Services/IEmailService.cs (hunk 3 above)
Comment (jpknoll) [reply Keas.Core/Services/IEmailService.cs]: You can specify .Text and .Html views on the Transaction.Content.

File: Keas.Core/Services/IEmailService.cs (hunk 2 above)
Comment (jSylvestre) [reply Keas.Core/Services/IEmailService.cs]: @jpknoll Needed, probably because the initial query has .AsNoTracking()

Comment (jpknoll) [reply Keas.Core/Services/IEmailService.cs]: Then we should remove the AsNoTracking, because we want to track them :-D

Comment (jSylvestre) [reply Keas.Core/Services/IEmailService.cs]: @jpknoll I think that caused a lot of other problems. But maybe @jcubbage knows for sure

**Pull request merged**